### PR TITLE
Feature/buildup govt inspection sheet

### PR DESCRIPTION
**Pull request by WernerVdM97**
Summary of this change:
Added a new govt inspection type, "buildup inspection". Buildup inspections have a `load_id` on the `govt_inspection_sheet` header. When a buildup inspection is created, all pallets with the flag `buildup_inspection_required` on the given load are included on the sheet.

CHANGELOG entry:

### Added
- `load_id` and `buildup_inspection `to `govt_inspection sheets`.
- Columns to `govt_inspection_pallets`:
  - `buildup_orig_inspection_passed`
  - `buildup_orig_inspected_at`
- Function `create_buildup_govt_inspection_pallets_for` to `govt_inspection_repo`
- Functionality to the TITAN inspection service to properly handle buildup inspections.

### Changed
- Moved `deduce_mixed_tm_params` from the govt inspection interactor to -repo.
- views, UI rules, interactor, routing and grid definitions to allow for buildup inspections.

## How to test this code via the UI:
Dependent on the first part of this feature Lux is developing, i.e. the buildup application changes.
Create a buildup inspection for a load with pallets with the flag `buildup_inspection_required` set to true.

Then proceed to create and inspect a buildup sheet.
| Finished Goods | Inspections | Govt Inspection Sheets | new Govt Buildup Inspection sheet

### CODE REVIEW

File: grid_definitions/lists/govt_inspection_sheets.yml [hunk 1]
```diff
@@ -41,4 +41,9 @@
   :url: "/finished_goods/inspection/govt_inspection_sheets/new/reinspection"
   :text: New Govt Reinspection Sheet
   :style: :button
-  :behaviour: :popup
```
Comment (WernerVdM97) [inline grid_definitions/lists/govt_inspection_sheets.yml:44]: There is a newline when viewed on my local machine and git detects it as up to date. 

Not sure why Git prompts this error and if it is actually present.

File: lib/finished_goods/interactors/govt_inspection_sheet_interactor.rb [hunk 2]
```diff
@@ -16,8 +21,10 @@ def create_govt_inspection_sheet(params) # rubocop:disable Metrics/AbcSize
 
       id = nil
       repo.transaction do
-        id = repo.create_govt_inspection_sheet(res)
+        id = repo.create_govt_inspection_sheet(res.to_h)
```
Comment (jcsjcs) [inline lib/finished_goods/interactors/govt_inspection_sheet_interactor.rb:24]: You shouldn't need to `to_h` here. The repo method should do it.

File: lib/finished_goods/repositories/govt_inspection_repo.rb [hunk 3]
```diff
@@ -124,10 +124,46 @@ def govt_inspection_sheet_packhouse_code(id)
       get_value(:plant_resources, :plant_resource_code, id: ph_id)
     end
 
+    def buildup_govt_inspection_sheet_for_load(load_id)
+      get_id(:govt_inspection_sheets, { load_id: load_id, buildup_inspection: true })
```
Comment (jcsjcs) [inline lib/finished_goods/repositories/govt_inspection_repo.rb:128]: Is there a reason for the curly braces?

File: lib/finished_goods/ui_rules/govt_inspection_sheet_rule.rb [hunk 4]
```diff
@@ -321,6 +338,7 @@ def add_controls # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity
     def add_behaviours
       behaviours do |behaviour|
         behaviour.input_change(:allow_tm_mixing, notify: [{ url: '/finished_goods/inspection/govt_inspection_sheets/allow_tm_mixing_checked' }])
+        behaviour.input_change(:buildup_inspection, notify: [{ url: '/finished_goods/inspection/govt_inspection_sheets/buildup_checked' }])
         behaviour.dropdown_change(:packed_tm_group_id, notify: [{ url: '/finished_goods/inspection/govt_inspection_sheets/packed_tm_group_changed' }])
         behaviour.dropdown_change(:destination_region_id, notify: [{ url: '/finished_goods/inspection/govt_inspection_sheets/destination_region_changed' }])
```
Comment (jcsjcs) [inline lib/finished_goods/ui_rules/govt_inspection_sheet_rule.rb:343]: These behaviours should ideally be changed to use the `handle_behaviour` pattern (run the scaffold to see hints in the comments.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

**Pull request merged**